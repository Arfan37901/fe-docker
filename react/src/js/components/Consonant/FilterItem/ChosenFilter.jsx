import React from 'react';
import PropTypes from 'prop-types';

const ChosenFilter = (props) => {
    const {
        name, id, parentId, onClick,
    } = props;

    const handleClick = () => {
        onClick(parentId, id, false);
    };

    return (
        <button
            type="button"
            onClick={handleClick}
            className="consonant-chosen-filter">{name}
        </button>
    );
};

export default ChosenFilter;

ChosenFilter.propTypes = {
    id: PropTypes.string.isRequired,
    parentId: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
};
