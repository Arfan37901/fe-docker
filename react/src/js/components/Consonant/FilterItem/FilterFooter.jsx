import React from 'react';
import PropTypes from 'prop-types';

const FilterFooter = (props) => {
    const {
        id,
        results,
        itemsSelected,
        onClearAll,
        clearFilterText,
        onClick,
    } = props;
    const handleClear = () => {
        onClearAll(id);
    };
    const handleClick = (clickEvt) => {
        clickEvt.preventDefault();
        onClick(id);
    };

    return (
        <div className="consonant-filter-footer">
            <span className="consonant-filter-footer--res-qty">{results} results</span>
            {itemsSelected > 0 &&
            <button
                type="button"
                onClick={handleClear}
                className="consonant-filter-footer--clear-btn">{clearFilterText}
            </button>}
            <button
                type="button"
                onClick={handleClick}
                className="consonant-filter-footer--btn">
                {itemsSelected > 0 ? 'Apply' : 'Done'}
            </button>
        </div>
    );
};

export default FilterFooter;

FilterFooter.propTypes = {
    id: PropTypes.string.isRequired,
    results: PropTypes.number.isRequired,
    itemsSelected: PropTypes.number,
    onClearAll: PropTypes.func.isRequired,
    onClick: PropTypes.func.isRequired,
    clearFilterText: PropTypes.string,
};

FilterFooter.defaultProps = {
    itemsSelected: 0,
    clearFilterText: 'Clear',
};
