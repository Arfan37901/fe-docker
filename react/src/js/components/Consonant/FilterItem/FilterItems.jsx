import React from 'react';
import PropTypes from 'prop-types';

const FilterItems = (props) => {
    const { id, items, onCheck } = props;
    const handleCheck = (evt) => {
        evt.stopPropagation();
        onCheck(id, evt.target.value, evt.target.checked);
    };

    return (
        <ul className="consonant-filter-items">
            {items.map(item => (
                <li
                    key={item.id}
                    className="consonant-filter-items--item">
                    <label className="consonant-filter-items--item-label">
                        <input
                            value={item.id}
                            type="checkbox"
                            onChange={handleCheck}
                            checked={item.selected} />
                        <span className="consonant-filter-items--item-checkmark" />
                        <span className="consonant-filter-items--item-name">{item.label}</span>
                    </label>
                </li>
            ))}
        </ul>
    );
};

export default FilterItems;

FilterItems.propTypes = {
    id: PropTypes.string.isRequired,
    onCheck: PropTypes.func.isRequired,
    items: PropTypes.arrayOf(PropTypes.object).isRequired,
};
