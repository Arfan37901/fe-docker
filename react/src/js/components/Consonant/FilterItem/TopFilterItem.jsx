import React from 'react';
import PropTypes from 'prop-types';

const TopFilterItem = (props) => {
    const {
        name,
        id,
        items,
        itemsSelected,
        isOpened,
        onCheck,
        onClick,
        onClearAll,
        results,
        clearFilterText,
    } = props;
    const handleClick = () => {
        onClick(id);
    };
    const handleClear = (clickEvt) => {
        clickEvt.stopPropagation();
        onClearAll(id);
    };
    const handleCheck = (evt) => {
        evt.stopPropagation();
        onCheck(id, evt.target.value, evt.target.checked);
    };
    const renderItems = () => (
        <ul className="consonant-top-filter--items">
            {items.map(item => (
                <li
                    key={item.id}
                    className="consonant-top-filter--items-item">
                    <label className="consonant-top-filter--items-item-label">
                        <input
                            value={item.id}
                            type="checkbox"
                            onChange={handleCheck}
                            checked={item.selected} />
                        <span className="consonant-top-filter--items-item-checkmark" />
                        <span className="consonant-top-filter--items-item-name">{item.label}</span>
                    </label>
                </li>
            ))}
        </ul>
    );
    const renderFooter = () => (
        <div className="consonant-top-filter--footer">
            <span className="consonant-top-filter--footer-res-qty">{results} results</span>
            {itemsSelected > 0 &&
            <button
                type="button"
                onClick={handleClear}
                className="consonant-top-filter--footer-clear-btn">{clearFilterText}
            </button>}
            <button
                type="button"
                onClick={handleClick}
                className="consonant-top-filter--footer-btn">
                {itemsSelected > 0 ? 'Apply' : 'Done'}
            </button>
        </div>
    );
    const defineClassNames = () => {
        const res = ['consonant-top-filter'];

        if (isOpened) res.push('consonant-top-filter_opened');
        if (items.filter(item => item.selected).length > 0 && !isOpened) res.push('consonant-top-filter_selected');

        return res.join(' ');
    };

    return (
        <div className={defineClassNames()}>
            <div className="consonant-top-filter--inner">
                <h3 className="consonant-top-filter--name">
                    <a href="#" className="consonant-top-filter--link" onClick={handleClick}>
                        {name}
                        <div className="consonant-top-filter--selcted-items-qty">
                            {items.filter(item => item.selected).length > 0 &&
                            items.filter(item => item.selected).length}
                        </div>
                    </a>
                </h3>
                <div className="consonant-top-filter--selcted-items">
                    <div className="consonant-top-filter--absolute-wrapper">
                        {renderItems()}
                        {renderFooter()}
                    </div>
                </div>
            </div>
        </div>
    );
};

export default TopFilterItem;

TopFilterItem.propTypes = {
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    onCheck: PropTypes.func.isRequired,
    onClick: PropTypes.func.isRequired,
    onClearAll: PropTypes.func.isRequired,
    items: PropTypes.arrayOf(PropTypes.object).isRequired,
    itemsSelected: PropTypes.number,
    isOpened: PropTypes.bool,
    results: PropTypes.number.isRequired,
    clearFilterText: PropTypes.string,
};

TopFilterItem.defaultProps = {
    isOpened: false,
    itemsSelected: 0,
    clearFilterText: 'Clear',
};
