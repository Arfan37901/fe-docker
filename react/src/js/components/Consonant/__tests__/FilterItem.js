import { screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import FilterItem from '../FilterItem';

import {
    DEFAULT_PROPS,
    selectedAllItems,
} from './constants/filterItem';

import makeSetup from './utils/settings';

const setup = makeSetup(FilterItem, DEFAULT_PROPS);

describe('Consonant/FilterItem', () => {
    test('should render without item count badge', () => {
        setup();

        const badgeElement = screen.queryByTestId('item-badge');

        expect(badgeElement).toBeNull();
    });
    test('should render with item count badge', () => {
        setup(selectedAllItems);

        const badgeElement = screen.queryByTestId('item-badge');

        expect(badgeElement).not.toBeNull();

        expect(badgeElement).toHaveTextContent(String(selectedAllItems.itemsSelected));
    });
    test('should render all list items', () => {
        const { props: { items } } = setup();

        const listItemElement = screen.queryAllByTestId('list-item');

        expect(listItemElement).toHaveLength(items.length);
    });

    describe('Check snapshots', () => {
        test('should renders correctly when closed', () => {
            const { tree } = setup();

            expect(tree).toMatchSnapshot();
        });
        test('should renders correctly when opened', () => {
            const { tree } = setup({ isOpened: true });

            expect(tree).toMatchSnapshot();
        });
    });

    describe('Interaction with UI', () => {
        test('should call onCheck', () => {
            const { props: { onCheck } } = setup();

            const [checkboxElement] = screen.queryAllByTestId('list-item-checkbox');

            expect(checkboxElement).toBeDefined();

            fireEvent.click(checkboxElement);

            expect(onCheck).toBeCalled();
        });
        test('should call onClick', () => {
            const { props: { onClick, name } } = setup();

            const itemLinkElement = screen.getByText(name);

            expect(itemLinkElement).not.toBeNull();

            fireEvent.click(itemLinkElement);

            expect(onClick).toBeCalled();
        });
        test('should call onClearAll', () => {
            const { props: { onClearAll } } = setup(selectedAllItems);

            const badgeElement = screen.queryByTestId('item-badge');

            expect(badgeElement).not.toBeNull();

            fireEvent.click(badgeElement);

            expect(onClearAll).toBeCalled();
        });
    });
});
