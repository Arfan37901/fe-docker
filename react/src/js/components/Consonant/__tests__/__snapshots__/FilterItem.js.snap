// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Consonant/FilterItem Check snapshots should renders correctly when closed 1`] = `
<div
  className="consonant-filters--item"
  data-testid="filter-item"
>
  <div
    className="consonant-filters--item-inner"
  >
    <h3
      className="consonant-filters--item-name"
    >
      <img
        alt=""
        loading="lazy"
        src="https://upload.wikimedia.org/wikipedia/commons/e/e0/Check_green_icon.svg"
        width="16"
      />
      <a
        className="consonant-filters--item-link"
        data-testid="filter-item__item-link"
        href="#"
        onClick={[Function]}
      >
        Topic
        <div
          className="consonant-filters--item-selcted-items"
          data-qty=""
        >
          
          
          
          
          
          
          
          
          
          
        </div>
      </a>
    </h3>
    <ul
      className="consonant-filters--item-list"
    >
      <li
        className="consonant-filters--item-list-item"
        data-testid="list-item"
      >
        <label
          className="consonant-filters--item-list-label"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            onChange={[Function]}
            type="checkbox"
            value="we-retail:activity/hiking"
          />
          <span
            className="consonant-filters--item-list-item-checkmark"
          />
          <span
            className="consonant-filters--item-list-item-name"
          >
            Business Continuity
          </span>
        </label>
      </li>
      <li
        className="consonant-filters--item-list-item"
        data-testid="list-item"
      >
        <label
          className="consonant-filters--item-list-label"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            onChange={[Function]}
            type="checkbox"
            value="we-retail:activity/biking"
          />
          <span
            className="consonant-filters--item-list-item-checkmark"
          />
          <span
            className="consonant-filters--item-list-item-name"
          >
            Creativity and Design
          </span>
        </label>
      </li>
      <li
        className="consonant-filters--item-list-item"
        data-testid="list-item"
      >
        <label
          className="consonant-filters--item-list-label"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            onChange={[Function]}
            type="checkbox"
            value="adobe-com-enterprise:topic/customer-intelligence"
          />
          <span
            className="consonant-filters--item-list-item-checkmark"
          />
          <span
            className="consonant-filters--item-list-item-name"
          >
            Customer Intelligence
          </span>
        </label>
      </li>
      <li
        className="consonant-filters--item-list-item"
        data-testid="list-item"
      >
        <label
          className="consonant-filters--item-list-label"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            onChange={[Function]}
            type="checkbox"
            value="adobe-com-enterprise:topic/data-management-platform"
          />
          <span
            className="consonant-filters--item-list-item-checkmark"
          />
          <span
            className="consonant-filters--item-list-item-name"
          >
            Data Management Platform
          </span>
        </label>
      </li>
      <li
        className="consonant-filters--item-list-item"
        data-testid="list-item"
      >
        <label
          className="consonant-filters--item-list-label"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            onChange={[Function]}
            type="checkbox"
            value="adobe-com-enterprise:topic/digital-foundation"
          />
          <span
            className="consonant-filters--item-list-item-checkmark"
          />
          <span
            className="consonant-filters--item-list-item-name"
          >
            Digital Foundation
          </span>
        </label>
      </li>
      <li
        className="consonant-filters--item-list-item"
        data-testid="list-item"
      >
        <label
          className="consonant-filters--item-list-label"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            onChange={[Function]}
            type="checkbox"
            value="adobe-com-enterprise:topic/digital-trends"
          />
          <span
            className="consonant-filters--item-list-item-checkmark"
          />
          <span
            className="consonant-filters--item-list-item-name"
          >
            Digital Trends
          </span>
        </label>
      </li>
      <li
        className="consonant-filters--item-list-item"
        data-testid="list-item"
      >
        <label
          className="consonant-filters--item-list-label"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            onChange={[Function]}
            type="checkbox"
            value="adobe-com-enterprise:topic/document-management"
          />
          <span
            className="consonant-filters--item-list-item-checkmark"
          />
          <span
            className="consonant-filters--item-list-item-name"
          >
            Document Management
          </span>
        </label>
      </li>
      <li
        className="consonant-filters--item-list-item"
        data-testid="list-item"
      >
        <label
          className="consonant-filters--item-list-label"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            onChange={[Function]}
            type="checkbox"
            value="adobe-com-enterprise:topic/marketing-automation"
          />
          <span
            className="consonant-filters--item-list-item-checkmark"
          />
          <span
            className="consonant-filters--item-list-item-name"
          >
            Marketing Automation
          </span>
        </label>
      </li>
      <li
        className="consonant-filters--item-list-item"
        data-testid="list-item"
      >
        <label
          className="consonant-filters--item-list-label"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            onChange={[Function]}
            type="checkbox"
            value="adobe-com-enterprise:topic/personalization"
          />
          <span
            className="consonant-filters--item-list-item-checkmark"
          />
          <span
            className="consonant-filters--item-list-item-name"
          >
            Personalization
          </span>
        </label>
      </li>
      <li
        className="consonant-filters--item-list-item"
        data-testid="list-item"
      >
        <label
          className="consonant-filters--item-list-label"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            onChange={[Function]}
            type="checkbox"
            value="adobe-com-enterprise:topic/Stock"
          />
          <span
            className="consonant-filters--item-list-item-checkmark"
          />
          <span
            className="consonant-filters--item-list-item-name"
          >
            Stock
          </span>
        </label>
      </li>
    </ul>
    <div
      className="consonant-filters--mobile-footer"
    >
      <span
        className="consonant-filters--mobile-footer-total-res"
      >
        0
         results
      </span>
      <button
        className="consonant-filters--mobile-footer-btn"
        onClick={[Function]}
        type="button"
      >
        Done
      </button>
    </div>
  </div>
</div>
`;

exports[`Consonant/FilterItem Check snapshots should renders correctly when opened 1`] = `
<div
  className="consonant-filters--item consonant-filters--item_opened"
  data-testid="filter-item"
>
  <div
    className="consonant-filters--item-inner"
  >
    <h3
      className="consonant-filters--item-name"
    >
      <img
        alt=""
        loading="lazy"
        src="https://upload.wikimedia.org/wikipedia/commons/e/e0/Check_green_icon.svg"
        width="16"
      />
      <a
        className="consonant-filters--item-link"
        data-testid="filter-item__item-link"
        href="#"
        onClick={[Function]}
      >
        Topic
        <div
          className="consonant-filters--item-selcted-items"
          data-qty=""
        >
          
          
          
          
          
          
          
          
          
          
        </div>
      </a>
    </h3>
    <ul
      className="consonant-filters--item-list"
    >
      <li
        className="consonant-filters--item-list-item"
        data-testid="list-item"
      >
        <label
          className="consonant-filters--item-list-label"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            onChange={[Function]}
            type="checkbox"
            value="we-retail:activity/hiking"
          />
          <span
            className="consonant-filters--item-list-item-checkmark"
          />
          <span
            className="consonant-filters--item-list-item-name"
          >
            Business Continuity
          </span>
        </label>
      </li>
      <li
        className="consonant-filters--item-list-item"
        data-testid="list-item"
      >
        <label
          className="consonant-filters--item-list-label"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            onChange={[Function]}
            type="checkbox"
            value="we-retail:activity/biking"
          />
          <span
            className="consonant-filters--item-list-item-checkmark"
          />
          <span
            className="consonant-filters--item-list-item-name"
          >
            Creativity and Design
          </span>
        </label>
      </li>
      <li
        className="consonant-filters--item-list-item"
        data-testid="list-item"
      >
        <label
          className="consonant-filters--item-list-label"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            onChange={[Function]}
            type="checkbox"
            value="adobe-com-enterprise:topic/customer-intelligence"
          />
          <span
            className="consonant-filters--item-list-item-checkmark"
          />
          <span
            className="consonant-filters--item-list-item-name"
          >
            Customer Intelligence
          </span>
        </label>
      </li>
      <li
        className="consonant-filters--item-list-item"
        data-testid="list-item"
      >
        <label
          className="consonant-filters--item-list-label"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            onChange={[Function]}
            type="checkbox"
            value="adobe-com-enterprise:topic/data-management-platform"
          />
          <span
            className="consonant-filters--item-list-item-checkmark"
          />
          <span
            className="consonant-filters--item-list-item-name"
          >
            Data Management Platform
          </span>
        </label>
      </li>
      <li
        className="consonant-filters--item-list-item"
        data-testid="list-item"
      >
        <label
          className="consonant-filters--item-list-label"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            onChange={[Function]}
            type="checkbox"
            value="adobe-com-enterprise:topic/digital-foundation"
          />
          <span
            className="consonant-filters--item-list-item-checkmark"
          />
          <span
            className="consonant-filters--item-list-item-name"
          >
            Digital Foundation
          </span>
        </label>
      </li>
      <li
        className="consonant-filters--item-list-item"
        data-testid="list-item"
      >
        <label
          className="consonant-filters--item-list-label"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            onChange={[Function]}
            type="checkbox"
            value="adobe-com-enterprise:topic/digital-trends"
          />
          <span
            className="consonant-filters--item-list-item-checkmark"
          />
          <span
            className="consonant-filters--item-list-item-name"
          >
            Digital Trends
          </span>
        </label>
      </li>
      <li
        className="consonant-filters--item-list-item"
        data-testid="list-item"
      >
        <label
          className="consonant-filters--item-list-label"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            onChange={[Function]}
            type="checkbox"
            value="adobe-com-enterprise:topic/document-management"
          />
          <span
            className="consonant-filters--item-list-item-checkmark"
          />
          <span
            className="consonant-filters--item-list-item-name"
          >
            Document Management
          </span>
        </label>
      </li>
      <li
        className="consonant-filters--item-list-item"
        data-testid="list-item"
      >
        <label
          className="consonant-filters--item-list-label"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            onChange={[Function]}
            type="checkbox"
            value="adobe-com-enterprise:topic/marketing-automation"
          />
          <span
            className="consonant-filters--item-list-item-checkmark"
          />
          <span
            className="consonant-filters--item-list-item-name"
          >
            Marketing Automation
          </span>
        </label>
      </li>
      <li
        className="consonant-filters--item-list-item"
        data-testid="list-item"
      >
        <label
          className="consonant-filters--item-list-label"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            onChange={[Function]}
            type="checkbox"
            value="adobe-com-enterprise:topic/personalization"
          />
          <span
            className="consonant-filters--item-list-item-checkmark"
          />
          <span
            className="consonant-filters--item-list-item-name"
          >
            Personalization
          </span>
        </label>
      </li>
      <li
        className="consonant-filters--item-list-item"
        data-testid="list-item"
      >
        <label
          className="consonant-filters--item-list-label"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            onChange={[Function]}
            type="checkbox"
            value="adobe-com-enterprise:topic/Stock"
          />
          <span
            className="consonant-filters--item-list-item-checkmark"
          />
          <span
            className="consonant-filters--item-list-item-name"
          >
            Stock
          </span>
        </label>
      </li>
    </ul>
    <div
      className="consonant-filters--mobile-footer"
    >
      <span
        className="consonant-filters--mobile-footer-total-res"
      >
        0
         results
      </span>
      <button
        className="consonant-filters--mobile-footer-btn"
        onClick={[Function]}
        type="button"
      >
        Done
      </button>
    </div>
  </div>
</div>
`;
